File: 0965-univalued-binary-tree/0965-univalued-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun isUnivalTree(root: TreeNode?): Boolean {
        if (root === null) {
            return true
        } else if (root.left !== null && root.left.`val` !== root.`val`) {
            return false
        } else if (root.right !== null && root.right.`val` !== root.`val`) {
            return false
        } else {
            return isUnivalTree(root.left) && isUnivalTree(root.right)
        }
    }
}

File: 0965-univalued-binary-tree/0965-univalued-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isUnivalTree = function(root) {
    return dfs(root, root.val)
    
    function dfs(root, val) {
        if (!root) return true
        if (root.val !== val) return false
        return dfs(root.left, val) && dfs(root.right, val)
    }
};